### PR TITLE
Lots More Concepts

### PR DESCRIPTION
**Pull request by timsehn**

### CODE REVIEW

File: content/concepts/dolt/git/working-set.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+On the command line, working set changes do follow to the newly checked out branch, just like Git. However, in SQL server mode, working set changes are not transferred to the newly checked out branch when you do a `dolt checkout`. The working set changes stay on the branch they were originally made. This change was made to account for multiple users using the same branch in SQL server mode.
+
+## Example
+
+### Make changes in a working set
+```
+docs $ dolt sql -q "insert into docs values (3,0)";
+Query OK, 1 row affected
+docs $ dolt status
```
Comment (max-hoffman) [inline content/concepts/dolt/git/working-set.md]:
```suggestion
$ dolt sql -q "insert into docs values (3,0)";
Query OK, 1 row affected
$ dolt status
```

File: content/concepts/dolt/git/working-set.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+On branch main
+Changes not staged for commit:
+  (use "dolt add <table>" to update what will be committed)
+  (use "dolt checkout <table>" to discard changes in working directory)
+	modified:       docs
+```
+
+### See what's changed in your working set
+```
+docs $ dolt diff
```
Comment (max-hoffman) [inline content/concepts/dolt/git/working-set.md]:
```suggestion
$ dolt diff
```

File: content/concepts/dolt/sql/indexes.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+# Databases
+
+## What is a Secondary Index?
+
+A secondary index can be added to any column or set of columns to convert lookup queries involving those columns into indexed lookups. Indexed lookups can be accomplished in constant time (ie. O(1)). 
+
+Secondary indexes are stored as separate data structures on disk or in memory. Thus, the use of secondary indexes uses more storage and increases insert and update time. 
+
+Secondary Indexes are called "secondary" to distinguish them from [primary keys](./primary-key.md) which also provide indexed lookups.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/indexes.md]:
```suggestion
Secondary indexes are called "secondary" to distinguish them from [primary keys](./primary-key.md) which also provide indexed lookups.
```

File: content/concepts/dolt/sql/table.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## What is a Table?
+
+Tables are the core unit of database [schema](./schema.md). Tables are defined by a set of columns. Columns can be [primary keys](./primary-key.md) which act as a unique identifier for each row. Once a table schema is defined, rows containing data can be inserted in to the table.
+
+Table data is stored on disk. The way a database lays out it's table data on disk defines some of the performance characteristics of the database. 
+
+## How to use Tables
+
+Structure the data you want to store and query in your database into tables. Define relationships between tables using foreign key [constraints](./constraints.md). Use `CREATE` statements to create tables and `ALTER` statements to change their definition.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/table.md]: something's off with this first sentence

Comment (max-hoffman) [inline content/concepts/dolt/sql/table.md]:
```suggestion
Structure the data you want to store and query in your database into tables. Define relationships between tables using foreign key [constraints](./constraints.md). Use `CREATE` statements to create tables and `ALTER` statements to change their schema.
```

File: content/concepts/dolt/sql/table.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Difference between MySQL Table and Dolt Table
+
+A MySQL and Dolt table are function the same on the surface. `CREATE` and `ALTER` statements work the same on both.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/table.md]:
```suggestion
A MySQL and Dolt table function the same on the surface. `CREATE` and `ALTER` statements work the same on both.
```

File: content/concepts/dolt/sql/table.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+Dolt and MySQL are [row major](https://en.wikipedia.org/wiki/Row-_and_column-major_order), meaning row values are stored next to each other as best as possible. However, MySQL stores data in a binary tree structure while Dolt stores table data on disk using a content-addressed binary tree called a [prolly tree](../../../architecture/storage-engine/prolly-tree.md). This setup makes Dolt [fairly comparable in query performance to MySQL](../../../reference/sql/benchmarks/latency.md) while also providing fast `diff` between versions. Fast `diff` powers Dolt's version control capabilities.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/table.md]:
```suggestion
Dolt and MySQL are [row major](https://en.wikipedia.org/wiki/Row-_and_column-major_order), meaning row values are stored next to each other. However, MySQL stores data in a binary tree structure while Dolt stores table data on disk using a content-addressed binary tree called a [prolly tree](../../../architecture/storage-engine/prolly-tree.md). This setup makes Dolt [fairly comparable in query performance to MySQL](../../../reference/sql/benchmarks/latency.md) while also providing fast `diff` between versions. Fast `diff` powers Dolt's version control capabilities.
```

Comment (max-hoffman) [inline content/concepts/dolt/sql/table.md]:
```suggestion
Dolt and MySQL are [row major](https://en.wikipedia.org/wiki/Row-_and_column-major_order), meaning row values are stored next to each other as best as possible. However, MySQL stores data in a binary tree structure while Dolt stores table data on disk using a content-addressed binary tree called a [prolly tree](../../../architecture/storage-engine/prolly-tree.md). This setup makes Dolt [fairly comparable in query performance to MySQL](../../../reference/sql/benchmarks/latency.md) while also providing history-independence and fast `diff` between versions. Fast `diff` powers Dolt's version control capabilities.
```

File: content/concepts/dolt/sql/transaction.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+# Transactions
+
+## What is a Transaction?
+
+A transaction is the unit of change isolation is a database. 
+
+## How to use Transactions
+
+Transactions are generally used to manage concurrent writes to the same data in a database. If two writers attempt to write the same data, one writer is allowed to commit their transaction and the other is rolled back. Changes being made in a transaction can only be seen by the current session.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/transaction.md:13]: maybe note that we're READ COMMITTED? (iirc)

File: content/concepts/dolt/sql/types.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+title: Types
+---
+
+# Types
+
+## What are Types?
+
+A column in a SQL database has a defined type, like an integer or a string. Some databases are more strict on types than others. MySQL supports [a number of types](https://dev.mysql.com/doc/refman/8.0/en/data-types.html).
+
+Types act as built in [constraints](./constraints.md) to help define the shape of your data. Are you expecting this column to be an integer or will it sometimes have decimal values? Defining the type explains to other database users what you expected when you defined the table.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/types.md:11]: And has a big impact on storage overhead and performance

File: content/concepts/dolt/sql/transaction.md (hunk 7 above)
Comment (timsehn) [reply content/concepts/dolt/sql/transaction.md:13]: We can add more later. are we READ COMMITTED?

Comment (max-hoffman) [reply content/concepts/dolt/sql/transaction.md:13]: REPEATABLE READ, my bad

File: content/concepts/dolt/sql/types.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+Moreover types act as a way to control the storage footprint of your database. A database must procure the maximum size of the type you define for each row when it stores your data on disk or memory.
+
+## How to use Types
+
+Each column in your tables must have a type. You use types when defining tables. You can change the type of a column with `ALTER` statements.
+
+When querying tables, the type of the column defines which operations can be used. For instance, a `>` (ie. greater than) operator can only be used on numerical types. To manipulate the type of a column when querying you use the `CAST()` function.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/types.md]: comparison operators work on all types? definitely also string types

File: content/concepts/dolt/sql/types.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+## Difference between MySQL Types and Dolt Types
+
+Dolt supports [a subset of MYSQL types](../../../reference/sql/sql-support/data-description.md#data-types) but otherwise shares the same behavior.
+
+## Interaction with Dolt Version Control
+
+A type change in Dolt is versioned and can cause conflicts when merged. Dolt can diff across some type changes and will make a best effort to do so. For some conversions, like string to blob, Dolt diff will error.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/types.md]: we should be able to compare between string and blob, probably worth a ticket if not

File: content/concepts/dolt/sql/views.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
+---
+title: Views
+---
+
+# Views
+
+## What is a View?
+
+Views look like tables when used in a select statement but the data in views is generated using SQL stored in the view definition. The data is actually stored in the tables the views reference not the view itself.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/views.md]:
```suggestion
Views look and act like tables, but the data in views is materialized on the fly using a view definition query that itself references concrete tables. The data is actually stored in the tables the views reference not the view itself.
```

Comment (max-hoffman) [reply content/concepts/dolt/sql/views.md]: not sure mine is strictly better, the first version is hard to read though

File: content/concepts/dolt/sql/views.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+## How to use Views
+
+Views allow you to derive tables using SQL instead of storing a copy of all the data you might want to derive. As a simple example, in a table of employee's salaries, you may store yearly salary but when using the table to calculate monthly salary, you use a view that divides the yearly salary by 12.
+
+## Difference between MySQL Views and Dolt Views
+
+There is no difference between MySQL and Dolt views. They are functionally equivalent.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/views.md:19]: i think mysql can materialize views?

File: content/concepts/dolt/sql/views.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+## Interaction with Dolt Version Control
+
+Dolt view definitions are versioned in the `dolt_schemas` system table. 
+
+If you would like to use a current view with an different version of the data, `as of` syntax works with views.
```
Comment (max-hoffman) [inline content/concepts/dolt/sql/views.md]: the current working set's view? this one wasn't completely obvious to me, so if the view is different in the other root we still use this root's view definition?

**Pull request merged**